src/LineageTree/lineageTree.py

```diff
@@ -2037,18 +2038,40 @@ def get_simple_tree(self, r: int, time_resolution: int = 1) -> tuple:
                 not the first time point).
             (dict) {m (int): duration (float)}: life time duration of the cell `m`
         """
+        def custom_successor(cell: int,time_end: int) -> int:
+            """Helper function that returns the length of a cell lifetime up to a certain timepoint,
+            if the cell lifetime ends before the time_end it returns the whole cycle.
+
+            Args:
+                cell (int): Id of the cell
+                time_end (int): The final timepoint
+
+            Returns:
+                (int): Length of the life cycle of a cell up to the time_end timepoint.
+            """
+            if self.time[cell] >time_end:
+                return 0
+            if self.time[self.get_cycle(cell)[-1]]>= time_end:
+                return time_end - self.time[cell]
+            if self.time[cell]==time_end:
+                return len(self.get_predecessors(cell))
+            else: return len(self.get_cycle(cell))
+
         if not hasattr(self, "cycle_time"):
             self.cycle_time = {}
         out_dict = {}
         to_do = [r]
         while to_do:
             current = to_do.pop()
-            cycle = self.get_successors(current)
-            _next = self.successor.get(cycle[-1], [])
+            cycle = custom_successor(current,time)
```

Comment:
This is a number of time points, not a cycle right??

Reply:
So , lets say I have a lineage treacked up to 500 timepoint and another up to 400 timepoint, so to make a correct comparison I either have to trim the .mastodon or make a cutoff. I chose the latter. Also I do not think it breaks to be honest.

Reply:
Ok, let's discuss that more specifically in a meeting, I am not sure our views align on the best way to do that :D

Reply:
Alright!